### PR TITLE
Enable testing with the Github Actions 

### PR DESCRIPTION
**Pull request by skirpichev**
On top of #554.

PS: This also seems to be a little faster than Travis-CI.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        pip install -U --force-reinstall setuptools pip
+        pip install -U .[develop]
```
Comment (cclauss) [inline .github/workflows/test.yml:18]:
```suggestion
        pip install --upgrade --force-reinstall setuptools pip
        pip install --upgrade .[develop]
```
Pip has both a `--upgrade` and a `--user option`.  Let's make it clear to the reader which one is happening.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      run: |
+        python -We:invalid -m compileall -f mpmath -q
+        pycodestyle
+    - name: Tests
+      run: py.test
+    - name: Coverage tests
+      env:
+        PYTEST_ADDOPTS: --cov mpmath
+      if: matrix.python-version == 2.7 || matrix.python-version == 3.6
+      run: py.test
```
Comment (cclauss) [inline .github/workflows/test.yml:29]:
```suggestion
      run: pytest
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    - name: Install dependencies
+      run: |
+        pip install -U --force-reinstall setuptools pip
+        pip install -U .[develop]
+    - name: Linting with pycodestyle, etc
+      run: |
+        python -We:invalid -m compileall -f mpmath -q
+        pycodestyle
+    - name: Tests
+      run: py.test
```
Comment (cclauss) [inline .github/workflows/test.yml:24]:
```suggestion
      run: pytest
```

**Review by cclauss — APPROVED** [review]

File: .github/workflows/test.yml (hunk 3 above)
Comment (cclauss) [reply .github/workflows/test.yml:24]: https://stackoverflow.com/questions/39495429/py-test-vs-pytest-command

**Comment by skirpichev** [comment]
FYI: Testing can be viewed in https://github.com/skirpichev/mpmath/actions for now.

**Comment by skirpichev** [comment]
@fredrik-johansson, can you review either this PR or its alternatives?  E.g. #554, #552.

CI for your project was broken long time ago, which is not good.  It's an easy to fix problem - lets fix this!

PS: Any news on #550?

**Pull request merged**